### PR TITLE
NMD-440 Added new get methods and tests for them, refactored update methods

### PR DESCRIPTION
**Pull request by glebtyanov**

### CODE REVIEW

File: SecondMap.Services.SMS/SecondMap.Services.SMS.UnitTests/Tests/Validators/Reviews/ReviewsUpdateValidatorTests.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+		{
+			_validator = new ReviewUpdateValidator();
+		}
+
+		[Theory]
+		[AutoMoqData]
+		public async Task Validate_WhenEveryFieldValid_ShouldReturnTrue(
+			ReviewUpdateViewModel validViewModel)
+		{
+			// Arrange
```
Comment (Glipchik) [inline SecondMap.Services.SMS/SecondMap.Services.SMS.UnitTests/Tests/Validators/Reviews/ReviewsUpdateValidatorTests.cs:20]: Remove

**Review by Glipchik — CHANGES_REQUESTED** [review]

File: SecondMap.Services.SMS/SecondMap.Services.SMS.BLL/Services/ScheduleService.cs [hunk 2]
```diff
@@ -73,5 +73,18 @@ public async Task DeleteScheduleAsync(int scheduleToDeleteId)
 
 			await _repository.DeleteAsync(entityToDelete);
 		}
+
+		public async Task<IEnumerable<Schedule>> GetAllByStoreIdAsync(int storeId)
+		{
+			var foundSchedules = await _repository.GetAllByPredicateAsync(sch => sch.StoreId == storeId);
+
+			if (!foundSchedules.Any())
```
Comment (moltenessence) [inline SecondMap.Services.SMS/SecondMap.Services.SMS.BLL/Services/ScheduleService.cs:81]: Isn't it better from the point of business logic and performance just to return empty collection and not to throw exception?

File: SecondMap.Services.SMS/SecondMap.Services.SMS.BLL/Services/StoreService.cs [hunk 3]
```diff
@@ -73,6 +73,19 @@ public async Task DeleteStoreAsync(int storeToDeleteId)
...
 			await _repository.DeleteAsync(entityToDelete);
 		}
+
+		public async Task<Store> GetByIdWithDetailsAsync(int id)
+		{
+			var foundStore = await _repository.GetByIdWithDetailsAsync(id);
+
+			if (foundStore == null)
+			{
+				Log.Error("Store with id = {@id} not found", id);
```
Comment (moltenessence) [inline SecondMap.Services.SMS/SecondMap.Services.SMS.BLL/Services/StoreService.cs:83]: Same

File: SecondMap.Services.SMS/SecondMap.Services.SMS.IntegrationTests/Tests/ReviewControllerTests.cs [hunk 4]
```diff
@@ -29,6 +32,40 @@ public async Task GetAll_WhenEntitiesExist_ShouldReturnSuccessAndDtoList()
...
+		[Fact]
+		public async Task GetAllByStoreId_WhenInvalidStoreId_ShouldReturnNotFound()
+		{
+			// Arrange
+			await _dataSeeder.CreateReviewAsync();
+			var invalidId = ValidationConstants.INVALID_ID;
+
+			// Act
+			var response =
+				await _client.GetAsync(TestConstants.REVIEWS_URL + "/" + ApiEndpoints.STORE_ID_EQUALS + $"{invalidId}");
```
Comment (moltenessence) [inline SecondMap.Services.SMS/SecondMap.Services.SMS.IntegrationTests/Tests/ReviewControllerTests.cs:63]: Implement path builder

File: SecondMap.Services.SMS/SecondMap.Services.SMS.BLL/Services/ScheduleService.cs (hunk 2 above)
Comment (glebtyanov) [reply SecondMap.Services.SMS/SecondMap.Services.SMS.BLL/Services/ScheduleService.cs:81]: ye, totally agree. thx

File: SecondMap.Services.SMS/SecondMap.Services.SMS.IntegrationTests/Tests/ReviewControllerTests.cs (hunk 4 above)
Comment (glebtyanov) [reply SecondMap.Services.SMS/SecondMap.Services.SMS.IntegrationTests/Tests/ReviewControllerTests.cs:63]: I'l make a separate pr to refactor tests later

**Review by Glipchik — APPROVED** [review]

**Review by Genresal — APPROVED** [review]

**Pull request merged**